### DISCUSSION TITLE
IPAM deployment with ACR

### DISCUSSION THREAD

**Discussion by kawan304** [Azure/ipam#277] [Q&A] [answered]
Hi -  I am trying to deploy IPAM with private container registry but getting the error show below while the deployment script is building an app container:

ERROR: The command failed with an unexpected error. Here is the traceback:
ERROR: 'charmap' codec can't encode character '\u2192' in position 3172: character maps to <undefined>

Anyone faced this issue?

**Comment by DCMattyG** [comment]
Hi @kawan304, the long story short is that the **apt-get update** process changed and now displays some Unicode characters that don't work so well in Windows machines and causes this issue.

This is being fixed in the next version which is coming out shortly, but here's a workaround for now:

1. Go to Settings
2. Click Time and language
3. Click Language and region
4. Click Administrative language settings
5. Click Change system locale
6. In the Region Settings dialog, select the Beta: Use Unicode UTF-8 for worldwide language support checkbox
7. Click OK to save any changes
8. Click Apply
9. Reboot your PC for the change to take effect

![image](https://github.com/Azure/ipam/assets/12955732/b6face11-c2fb-45af-9e5f-56ebcff65f1d)

Hope that helps for now, new version will be here shortly!

**Reply by kawan304** [reply]
Thanks, DCMattyG - That worked, and the installation was successful without any errors. But now I am seeing following errors in the logs and I get application error in the browser when I browse the webapp.

2024-06-14T03:04:40.165Z ERROR - Container ipam-77szlgkslwpvc_0_58761a65 for site ipam-77szlgkslwpvc did not start within expected time limit. Elapsed time = 230.9076881 sec
2024-06-14T03:04:40.170Z ERROR - Container ipam-77szlgkslwpvc_0_58761a65 didn't respond to HTTP pings on port: 80, failing site start. See container logs for debugging.
2024-06-14T03:04:40.183Z INFO - Stopping site ipam-77szlgkslwpvc because it failed during startup.

**Comment by DCMattyG** [comment]
Hi @kawan304, I believe this is due to an issue I found in the current deployment script where it sets the listening port of the container to 80, which isn't supported.

To fix it, re-deploy the container by executing these commands from the root directory of the ipam folder:

1. **az login**
2. **az account set -s &lt;Subscription ID Where IPAM is Deployed&gt;**
3. **az acr build -r &lt;Private ACR Name&gt; -t ipam:latest -f ./Dockerfile.deb .**

> NOTE: These instructions can also be found [here](https://azure.github.io/ipam/#/contributing/README?id=building-amp-updating-production-containers-images-using-a-private-acr) on our docs page.

That should properly build the container and push it to the ACR. After that restart your App Service and give it a few minutes to pull down the new container.

Hope that helps, and the new code should fix all of this very shortly!

**Comment by DCMattyG** [comment, marked as answer]
Hey there @kawan304, I just wanted to let you know that these issue have been resolved with PR #279 and are completely addressed in the latest release v3.2.0.

Please let me know if you're still have issues and I'll be happy to assist however I can.

**Reply by kawan304** [reply]
Thanks a lot.

Deployed successfully :).

**Reply by DCMattyG** [reply]
Wonderful to hear, thanks for checking!
